include/picongpu/plugins/binning/axis/LinearAxis.hpp

```diff
@@ -0,0 +1,256 @@
+                    /** Enable or disable allocation of extra bins for out of range particles*/
+                    bool overflowEnabled;
+
+                    constexpr LinearAxisKernel(T_AttrFunctor attrFunc, bool enableOverflowBins)
+                        : getAttributeValue{attrFunc}
+                        , overflowEnabled{enableOverflowBins}
+                    {
+                    }
+
+                    /**
+                     * @param n_bins User requested n_bins, actual nBins is n_bins + 2 for overflow and underflow bins
+                     */
+                    void initAxisSplit(T minR, T maxR, uint32_t n_bins, ScalingType scalingR)
+                    {
+                        min = minR;
+                        max = maxR;
+                        nBins = n_bins;
+                        scaling = scalingR;
+                    }
+
+                    template<typename T_Worker, typename T_Particle>
+                    ALPAKA_FN_ACC uint32_t getBinIdx(
+                        const DomainInfo& domainInfo,
+                        const T_Worker& worker,
+                        const T_Particle& particle,
+                        bool& validIdx) const
+                    {
+                        auto val = getAttributeValue(domainInfo, worker, particle);
+
+                        static_assert(
+                            std::is_same<decltype(val), decltype(min)>::value,
+                            "The return type of the axisAttributeFunctor should be the same as the type of Axis "
+                            "min/max ");
+                        uint32_t binIdx = 0;
+                        bool enableBinning = overflowEnabled; // @todo check if disableBinning is better
+                        // @todo check for optimizations here
+                        if(val >= min)
+                        {
+                            if(val < max)
+                            {
+                                /**
+                                 * Precision errors?
+                                 * Is the math floor necessary?
+                                 */
+                                binIdx = math::floor(((val - min) * scaling) + 1);
+                                if(!overflowEnabled)
+                                {
+                                    enableBinning = true;
+                                    binIdx = binIdx - 1;
+                                }
+                            }
+                            else
+                                binIdx = nBins - 1;
+                        }
+                        validIdx = validIdx && enableBinning;
+                        return binIdx;
+                    }
+                };
+
+                LinearAxisKernel lAK;
+
+                struct BinWidthKernel
+                {
+                    ScalingType scaling;
+                    uint32_t nBins;
+
+                    BinWidthKernel()
```

Comment:
```suggestion
                    BinWidthKernel() = default;
```

But do we really need an empty constructor? `BinWidthKernel` is not in a usable state afterwards due to un-initialised memory. But even if we fix that, does it make sense, to have `scaling` and `nBins` on some (likely mismatching with other parts) default value?

Reply:
Removed the empty constructor